metrics/prometheus.go

```diff
@@ -33,6 +33,11 @@ import (
 // asFloat64 converts a uint64 into a float64.
 func asFloat64(v uint64) float64 { return float64(v) }
 
+// asMicrosecondsToSeconds converts nanoseconds into a float64 representing seconds.
+func asMicrosecondsToSeconds(v uint64) float64 {
+	return float64(v) / float64(time.Millisecond)
```

Comment:
I wonder if we should not use `time` constants, as this is slightly confusing having `time.Millisecond` here.

```suggestion
	return float64(v) / 1e6
```

Reply:
Seems sensible, I've updated both functions.